v1/server/server.go

```diff
@@ -2677,31 +2726,44 @@ func (s *Server) updateNDCache(enabled bool) {
 	s.ndbCacheEnabled = enabled
 }
 
-func stringPathToDataRef(s string) (r ast.Ref) {
+func stringPathToDataRef(s string) (ast.Ref, error) {
 	result := ast.Ref{ast.DefaultRootDocument}
-	return append(result, stringPathToRef(s)...)
+	r, err := stringPathToRef(s)
+	if err != nil {
+		return nil, err
+	}
+	return append(result, r...), nil
 }
 
-func stringPathToRef(s string) (r ast.Ref) {
+func stringPathToRef(s string) (ast.Ref, error) {
+	r := ast.Ref{}
+
 	if len(s) == 0 {
-		return r
+		return r, nil
 	}
+
 	p := strings.Split(s, "/")
 	for _, x := range p {
 		if x == "" {
 			continue
 		}
+
 		if y, err := url.PathUnescape(x); err == nil {
 			x = y
 		}
+
+		if strings.ContainsAny(x, "\"") {
```

Comment:
This is a simple check that the ref component doesn't contain a `"` that would terminate it's string representation. We could be way more restrictive here; but then we also do have that other check asserting that we don't produce more than a single ref in the ultimate query.